### ISSUE TITLE
genericMatrix should be row major

### ISSUE THREAD

**Issue by mahrud** [Macaulay2/M2#3596]
It was pointed out during the workshop that the output of `genericMatrix` should be row major instead of column major because this makes no sense:
```m2
i1 : R = QQ[x_(1,1) .. x_(2,5)];

i2 : genericMatrix(R, 2, 5)

o2 = | x_(1,1) x_(1,3) x_(1,5) x_(2,2) x_(2,4) |
     | x_(1,2) x_(1,4) x_(2,1) x_(2,3) x_(2,5) |

             2      5
o2 : Matrix R  <-- R
```

**Comment by d-torrance** [comment]
This was bugging me the other day, too.  I wonder how much code it would break if we changed it?  For example, `permanents` fixes the problem by doing this:

```m2
M1:= transpose genericMatrix(R1,xxX_(1,1),c,r);
```

**Comment by mahrud** [comment]
I personally think it's a change worth trying, and if we find that it breaks way too many things we can just add `genericMatrix'` which does the transpose.

**Comment by mikestillman** [comment]
I agree that the proposed way is better, but I'm afraid it will break alot of code.  Not code in the system so much, but in particular, for personal code of mine for different projects of mine.

**Comment by pzinn** [comment]
I remember this as being one of the first things that annoyed me when I started using M2 twenty years ago. And adding transpose is not great because the degree shift becomes visible
```
i2 : transpose genericMatrix(QQ[m_(1,1)..m_(2,2)],2,2)

o2 = {-1} | m_(1,1) m_(1,2) |
     {-1} | m_(2,1) m_(2,2) |

                            2                     2
o2 : Matrix (QQ[m   ..m   ])  <-- (QQ[m   ..m   ])
                 1,1   2,2             1,1   2,2
```

**Comment by mahrud** [comment]
Yes, I was also gonna mention the degree change.

What if we add a new method, say `generic(Module, Module)`, which produces the new output, and have `genericMatrix` give a deprecation warning from now on?

**Comment by mahrud** [comment]
This is also bad:
```
i4 : R = QQ[x_(1,1)..x_(2,3)];

i5 : reshape(R^2, R^3, vars R)

o5 = | x_(1,1) x_(1,3) x_(2,2) |
     | x_(1,2) x_(2,1) x_(2,3) |

             2      3
o5 : Matrix R  <-- R
```

**Comment by pzinn** [comment]
maybe add an option `RowMajor` to `genericMatrix`? also at some point I experimented with an alternative syntax `R=QQ[m_(1,1)...m_(5,5); M=genericMatrix m` where M2 figures out on its own the correct shape.

**Comment by mahrud** [comment]
I just have a `genericMatrix'` in my scripts that does the right thing, but an option is fine too.
